src/classes/Expect.cls

```diff
@@ -57,47 +57,47 @@ public class Expect {
         }
 
         public void shouldEqual(Object actual, String message) {
-            assert(expected == actual, message);
+            assert(JSON.serialize(expected).equals(JSON.serialize(actual)), message);
```

Comment:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_expressions_operators_understanding.htm

=== is a thing to know about here.

http://peterknolle.com/apex-equals/
Objects answer if they are equal to each other when asked... which is why the list behaved in the unexpected way.

Looking to other existing libraries for guidance we have terms like equals, strictEquals, equivalence to sort out the different types of equality

Serialize is not the way out. I didn't test this code b/c I'm about to head out but I think you could see why this would break the shouldEquals based on serialization. 
```java
class Foo {
  Integer i = 0;
  public Integer prop {
     get {
         i = i + 1;
         return i;
     }
}
```